app/javascript/react/components/InvitationBlock.jsx

```diff
@@ -17,21 +18,33 @@ export default function InvitationBlock({
   status,
 }) {
   const [isLoading, setIsLoading] = useState({
-    smsInvitation: false,
-    emailInvitation: false,
-    postalInvitation: false,
+    sms: false,
+    email: false,
+    postal: false,
   });
-  const [lastSmsInvitationSentAt, setLastSmsInvitationSentAt] = useState(
-    retrieveLastInvitationDate(invitations, "sms")
+  /* eslint no-unused-vars: ["error", { "varsIgnorePattern": "setInvitationsDatesByFormat" }] */
+  // This state is updated by .unshift in handleInvitationClick method
+  const [invitationsDatesByFormat, setInvitationsDatesByFormat] = useState(
+    sortInvitationsByFormatsAndDates(invitations)
   );
-  const [lastEmailInvitationSentAt, setLastEmailInvitationSentAt] = useState(
-    retrieveLastInvitationDate(invitations, "email")
-  );
-  const [lastPostalInvitationSentAt, setLastPostalInvitationSentAt] = useState(
-    retrieveLastInvitationDate(invitations, "postal")
+  const [showInvitationsHistory, setShowInvitationsHistory] = useState(false);
+
+  // We don't display "Show history" button if there is no history to display
+  const showInvitationsHistoryButton = Object.values(invitationsDatesByFormat).some(
+    (array) => array.length > 1
   );
 
-  const showInvitation = (format) => invitationFormats.includes(format);
+  const computeInvitationsDatesRows = () => {
+    const numberOfInvitationsDatesRowsNeeded = Object.values(invitationsDatesByFormat).reduce(
+      (r, s) => (r > s.length ? r : s.length),
+      0
+    );
+    const invitationsDatesRowsArray = [];
+    for (let i = 0; i < numberOfInvitationsDatesRowsNeeded; i += 1) {
+      invitationsDatesRowsArray.push(i);
+    }
+    return invitationsDatesRowsArray;
+  };
```

Comment:
Je suis pas sûr de comprendre cette méthode (qui par son nom indiquerait qu'on compute les lignes directement).
Est-ce qu'on veut pas juste le nombre de ligne ?
Dans ce cas là c'est juste le max de l'array `invitationFormats.map(format => invitationsDatesByFormat[format].length)` non ?

Reply:
Dans un premier temps on veut le nombre de ligne ; c'est ce que je calcule dans `numberOfInvitationsDatesRowsNeeded `. On peut effectivement le faire avec un map + un Math.max(), quand on additionne les deux je crois que c'est légèrement moins performant qu'un reduce mais bon de toute façon ça ne change pas grand chose sur des arrays aussi petites. Je vais mettre ta solution qui est peut-être plus simple à lire.

En revanche dans un deuxième temps je veux un array du nombre de lignes. Par exemple, pour 3 lignes, je veux `[0, 1, 2]` pour ensuite pouvoir itérer dessus en utilisant le chiffre pour l'index et la key du component `InvitationsDatesRow`. Mais il y a peut-être plus simple

Reply:
Ok, j'ai trouvé plus simple, j'ai refactoré.

Reply:
Oui je comprends ce que tu veux dire, y a pas le type `Range` en javascript qui permet de faire ça facilement comme en ruby. Mais tu peux construire l'array facilement en en faisant `[...Array(n).keys()]`.
Là c'est un peu bizarre ce que tu fais, tu fabriques un array de components dans une fonction et tu appelles cette fonction quand tu render. 
En vrai ça marche, mais ça veut peut-être dire qu'il manque une abstraction (un component `InvitationDatesRows` comme on le fait dans `ApplicantList`?).